cmd/mobileproxy/mobileproxy.go

```diff
@@ -0,0 +1,151 @@
+// Package mobileproxy instantiates a Martian Proxy (go/martianproxy).
+// This package is a reference implmentation of Martian Proxy intended to
+// be cross compiled with google3/go/tools/gobind for use on Android and iOS.
+package mobileproxy
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"flag"
+	"github.com/google/martian"
+	"github.com/google/martian/har"
+	"github.com/google/martian/httpspec"
+	mlog "github.com/google/martian/log"
+	"github.com/google/martian/martianhttp"
+	"github.com/google/martian/mitm"
+	"github.com/google/martian/verify"
+	"log"
+	"net"
+	"net/http"
+	"time"
+
+	// side-effect importing modifiers to register them with the proxy
+	_ "github.com/google/martian/body"
+	_ "github.com/google/martian/cookie"
+	_ "github.com/google/martian/fifo"
+	_ "github.com/google/martian/header"
+	_ "github.com/google/martian/martianurl"
+	_ "github.com/google/martian/method"
+	_ "github.com/google/martian/pingback"
+	_ "github.com/google/martian/priority"
+	_ "github.com/google/martian/querystring"
+	_ "github.com/google/martian/skip"
+	_ "github.com/google/martian/status"
+)
+
+// Martian is a wrapper for the initialized Martian proxy.
+type Martian struct {
+	proxy    *martian.Proxy
+	listener net.Listener
+	mux      *http.ServeMux
+}
+
+// Start runs a martian.Proxy on addr.
+func Start(proxyAddr string) (*Martian, error) {
+	return StartWithCertificate(proxyAddr, "", "")
+}
+
+// StartWithCertificate runs a proxy on addr and configures a cert for MITM
+func StartWithCertificate(proxyAddr string, cert string, key string) (*Martian, error) {
```

Comment:
@brandonnelson3 You kind of have to twist your thinking a bit for this since the API defined here is the one exposed in Java via gobind. I'm not sure how the tools deal with []byte and Java users will probably just read the cert/key files into a string anyway.

Same thing with the `Listener`. It's too much surface area exposed on the Java side and may just end up breaking `gobind`. We are kind of bound by what it supports and the last I checked it is very limited in what it can export.